filters/apiusagemonitoring/filter_endpointmetrics_test.go

```diff
@@ -390,8 +391,10 @@ func Test_Filter_String(t *testing.T) {
 	if !assert.IsType(t, &apiUsageMonitoringFilter{}, filter) {
 		return
 	}
-	f := filter.(*apiUsageMonitoringFilter)
-	s1 := fmt.Sprintf("%s", f)
-	s2 := fmt.Sprintf("%s", *f)
-	assert.Equal(t, s1, s2)
+	f, ok := filter.(*apiUsageMonitoringFilter)
```

Comment:
Shouldn't there now be an else clause for the failing test?

Reply:
ah sorry, my fault I tried to hack around the Sprintf, because staticcheck found that we should use String(), but String() call MarshalJSON, which does a nil ptr dereference in some test cases. The code was too weird to me, but maybe required, have to understand better the api monitoring code, before I understand how to workaround the problem.

Reply:
I changed it back to the original version